main.cpp

```diff
@@ -334,6 +256,38 @@ FloatType::~FloatType()
     delete value;
 }
 
+FloatType& FloatType::pow(float num)
+{
+    powInternal(num);
+    return *this;
+}
+
+FloatType& FloatType::pow(const FloatType& ftype)
+{
+    powInternal(ftype);
+    return *this;
+}
+
+FloatType& FloatType::pow(const IntType& ftype)
+{
+    float arg = static_cast<float>(ftype);
+    powInternal(arg);
+    return *this;
+}
+
+FloatType& FloatType::pow(const DoubleType& ftype)
+{
+    float arg = static_cast<float>(ftype);
+    powInternal(arg);
+    return *this;
```

Comment:
What does `powInternal` return?
Why aren't you returning what that function returns?
You should, doing so will eliminate a few lines of repetitive code.
- [ ] revise